CIP-0159/README.md

```diff
@@ -0,0 +1,587 @@
+`AccountValue` to the new `AccountValue`, the transaction must specify direct deposits for those
+assets in the same transaction. The assets must effectively be re-deposited.
+- If you are only using the `unreg_account_value_cert`, you will be able to recover the part of the
+  protocol deposit that was for the `AccountValue`. **The account address itself remains
+registered for delegation.**
+- Transactions involving these certificates must be witnessed by the associated staking credentials.
+  These certificates must be approved by the community to be used on the Cardano Treasury account
+address.
+
+For this to work, the transaction representation needs a new direct deposit field:
+
+```cbor
+direct_deposits = {+ reward_account => value}
+
+transaction_body = 
+  {   0  : set<transaction_input>         
+  ,   1  : [* transaction_output]      
+  ,   2  : coin                            
+  , ? 3  : slot_no                         
+  , ? 4  : certificates                    
+  , ? 5  : withdrawals                     
+  ...
+  , ? 23 : direct_deposits ; new field
+  }
+```
+
+> [!TIP]
+> *Why not just make withdrawals support negative numbers?*
+>
+> 1. It would cut in half the maximum supported number.
+> 2. Withdrawals require witnesses while direct deposits do not. This means we need to be able to
+>    distinguish between direct deposits and withdrawals with for the `ScriptPurpose`.
+>
+> Adding a separate field is the better approach.
+
+Plutus scripts will be able to see the direct deposits occuring in the transaction as part of their
+`ScriptContext`. See the [New Plutus Script Context section](#new-plutus-script-context).
+
+> [!IMPORTANT]
+> The direct deposits are only the diff! If the account currently has 100 ADA and the transaction is
+> direct depositing another 0.1 ADA, the transaction only needs to specify 0.1 ADA.
+
+### Partial Withdrawals and Native Asset Withdrawals
+
+The ledger rules *must* be changed to allow withdrawing any amount between 0 and the current
+balance. 
+
+> [!IMPORTANT]
+> For all withdrawals, partial or full, the pre-existing `Withdraw` purpose will be used. We do not
+> need a dedicated purpose for partial withdrawals.
+
+To enable support for withdrawing native assets from account addresses, the representation of
+withdrawals inside transactions needs to be updated:
+
+```cbor
+; Old representation.
+withdrawals = {+ reward_account => coin}
+
+; New representation.
+withdrawals = {+ reward_account => value} ; replaced `coin` with `value`
+```
+
+Likewise, since the Cardano Treasury is also able to hold whitelisted native assets, the
+`treasury_withdrawals_action` must be updated:
+
+```cbor
+; Old representation.
+treasury_withdrawals_action = (2, {* reward_account => coin}, policy_hash/ nil)
+
+; New representation.
+treasury_withdrawals_action = (2, {* reward_account => value}, policy_hash/ nil) ; replaced `coin` with `value`
+```
```

Comment:
I think this feature of holding native assets in a treasury is best left for another CIP. They could be implemented at the same time, but there are plenty of other design issues that will need to be resolved before we allow multi-assets in a treasury. This CIP, by far is the biggest pre-requisit, but it is not the only problem that will need to be solved. So, I suggest we remove this part of the CIP
```suggestion
```